src/components/battles/animations/shared/Fainting.ts

```diff
@@ -39,18 +39,22 @@ export class Fainting<TGameStartr extends FullScreenPokemon> extends Component<T
             this.gameStarter.physics.getMidY(thing) + thing.height * thing.scale!,
             1,
             (): void => {
+                const playerName = this.gameStarter.itemsHolder.getItem("name");
+                let partyIsWipedText: (string | string[])[][] = [[pokemon.nickname, " fainted!"]];
                 this.gameStarter.physics.killNormal(thing);
                 this.gameStarter.physics.killNormal(blank);
                 this.gameStarter.menuGrapher.createMenu("GeneralText");
-                this.gameStarter.menuGrapher.addMenuDialog(
-                    "GeneralText",
-                    [
+                if (this.gameStarter.battles.isPartyWiped) {
+                    partyIsWipedText = [
```

Comment:
Instead of reassigning, push to the existing array. That'll let you make it `const`.